Task_2/CarCollection.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task_2
+{
+    public class CarCollection<T> : ICarCollection<T>
+    {
+        private T[] myCarCollection;
+
+        public CarCollection()
+        {
+            this.myCarCollection = new T[0];
+            this.Count = 0;
+        }
+
+        public CarCollection(int length)
+        {
+            this.myCarCollection = new T[length];
+            this.Count = length;
+        }
+
+        public T this[int index]
+        {
+            get
+            {
```

Comment:
What about validation if internal array does not contain the specified index?

Reply:
But according to the task description we should have implemented the index by car name.